pyveg/src/analysis_preprocessing.py

```diff
@@ -42,34 +42,36 @@ def read_json_to_dataframes(data):
 
         rows_list = []
 
-        # loop over time series
-        for date, time_point in coll_results["time-series-data"].items():
+        if "time-series-data" in coll_results.keys():
 
-            # check we have data for this time point
-            if time_point is None or time_point == {} or time_point == []:
+            # loop over time series
+            for date, time_point in coll_results["time-series-data"].items():
 
-                # add Null row if data is missing at this time point
-                rows_list.append({"date": date})
+                # check we have data for this time point
+                if time_point is None or time_point == {} or time_point == []:
```

Comment:
i think all of this can be written as 
```
if not time_point
```
(empty lists and dicts count as "None")